rubyforgood/diaper
# [#2159] Display the Approve Partner button on the top with a tooltip

Resolves #2159

### Description
In the Approve Partner screen, the Approve Partner button was at the bottom of the page.
Since the page is quite long, it would be interesting if the user could see the button either at the top or while they're scrolling the page.

Two options were suggested by @edwinthinks. The first is placing the button at the top of the page, beside the "Application Details" header. However, this may encourage the user to not read the application instead and just click Approve. The other option would be to add a fixed button that scrolls with the user, which is what I did.



To test the changes I also added a fixture to help mock the Partner API.

### Type of change



* Minor adjustment (non-breaking change which fixes an non-bug issue)

### How Has This Been Tested?

#### Manual testing:
* After setting up both Diaper and Partner, log in Diaper, go to Partner Agencies
* Add two Partner agencies and invite both of them
* Log in the Partner app with one of the partners and request a review
* In the Diaper app, there should be at least two Partners now: one invited and one awaiting review, both of which should have the "Review Application" action available
* Click on the Review Application button of the "invited" partner. The Approve Partner button should be visible but disabled, and it should show a tooltip informing the user what to do.
* Go back and click on the Review Application of the "awaiting review" partner. The Approve Partner button should be visible, enabled and should display no tooltip.

#### Automated testing:
For each of the partners above (one invited and one awaiting review) I added the following tests:
* Test if the Approve Partner button is always visible
* Test if the button is enabled for the "awaiting review" partner, but disabled for the "invited" partner
* Test if the button displays a tooltip for the "invited" partner, but shows no tooltip for the "awaiting review" partner



### Screenshots


#### Before changes
The Approve Partner button was at the bottom of the page, only visible and enabled for "awaiting review" partners and had no tooltip.

The button was never visible when the user was at the top of the page.
![before-top-of-the-page](https://user-images.githubusercontent.com/72531802/108511539-d2aec380-729e-11eb-961e-48a442afad2d.png)

In case the user scrolled down, viewing a "invited" partner, the button was not visible:
![before-bottom-of-the-page-invisible](https://user-images.githubusercontent.com/72531802/108511657-f96cfa00-729e-11eb-8286-6376dd603d59.png)

In case the user scrolled down, viewing an "awaiting review" partner, the button was visible:
![before-bottom-of-the-page](https://user-images.githubusercontent.com/72531802/108511788-215c5d80-729f-11eb-9417-0b476ab6ae2e.png)

#### After changes
The button is now at a fixed position at the bottom right of the screen. The button is always visible, but disabled for "invited" partners. It also displays a tooltip for "invited" partners.

Desktop view:
![screenshot-desktop](https://user-images.githubusercontent.com/72531802/108511840-32a56a00-729f-11eb-8514-32a93ce4cca3.png)

When scrolling to the bottom of the page:
![screenshot-scroll-bottom](https://user-images.githubusercontent.com/72531802/108512282-beb79180-729f-11eb-98c4-e96b46206000.png)

Mobile view:
![screenshot-mobile](https://user-images.githubusercontent.com/72531802/108511852-3638f100-729f-11eb-85cc-37062b2012ce.png)

Tooltip:
![disabled-tooltip](https://user-images.githubusercontent.com/72531802/108511866-3b963b80-729f-11eb-838e-2dfc6415bcb5.png)
